src/lib/libast/tests/path/pathpath.c

```diff
@@ -0,0 +1,46 @@
+#include "config_ast.h"  // IWYU pragma: keep
+
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "ast.h"
+#include "terror.h"
+
+char test_path[PATH_MAX] = "foo";
+void cleanup_handler(void) { unlink(test_path); }
+
+tmain() {
+    UNUSED(argc);
+    UNUSED(argv);
+
+    char path[PATH_MAX];
+    char absolute_test_path[PATH_MAX];
+
+    getcwd(absolute_test_path, PATH_MAX);
+    strcat(absolute_test_path, "/");
+    strcat(absolute_test_path, "foo");
+
+    int fd = open(test_path, O_CREAT, 0666);
+    if (fd == -1) terror("Failed to create test file");
+    atexit(cleanup_handler);
+
+    // Search this path in current directory and expand it
+    pathpath("foo", NULL, PATH_ABSOLUTE, path, PATH_MAX);
+    if (strcmp(path, absolute_test_path))
+        terror("Failed to expand path of file in current directory");
+
+    if (!pathpath("cat", NULL, PATH_EXECUTE, path, PATH_MAX))
+        terror("Failed to find `cat` in current $PATH");
+
+    if (!pathpath(path, NULL, PATH_EXECUTE, path, PATH_MAX))
```

Comment:
Passing `path` twice like this causes this statement at line 65 to fail outright on macOS due to overlapping buffers and causes undefined behavior on every platform:

```
strcpy(path, path_to_search);
```